dot/sync/chain_sync.go

```diff
@@ -503,12 +516,38 @@ func (cs *chainSync) getTarget() *big.Int {
 		return big.NewInt(2<<32 - 1)
 	}
 
+	// we are going to sort the data and remove the outliers then we will return the avg of all the valid elements
+	intArr := make([]interface{}, len(cs.peerState))
 	for _, ps := range cs.peerState {
-		sum = big.NewInt(0).Add(sum, ps.number)
+		intArr = append(intArr, *ps.number)
+	}
+	//now sort the array
+	sort.Sort(BigIntSorter(intArr))
+
+	reducerSum := func(a, b interface{}) interface{} {
 		count++
+		return big.NewInt(0).Add(a.(*big.Int), b.(*big.Int))
+	}
+
+	comp := func(a, b interface{}) int {
+		return a.(*big.Int).Cmp(b.(*big.Int))
+	}
+
+	plus := func(a, b interface{}) interface{} {
+		return big.NewInt(0).Add(a.(*big.Int), b.(*big.Int))
+	}
+	minus := func(a, b interface{}) interface{} {
+		return big.NewInt(0).Sub(a.(*big.Int), b.(*big.Int))
+	}
+	divide := func(a, b interface{}) interface{} {
+		return big.NewInt(0).Div(a.(*big.Int), big.NewInt(int64(b.(int))))
+	}
+	mul := func(a, b interface{}) interface{} {
+		return big.NewInt(0).Mul(a.(*big.Int), big.NewInt(int64(b.(float64))))
 	}
+	sum := RemoveOutlier(intArr, comp, big.NewInt(0), reducerSum, plus, minus, divide, mul).(*big.Int)
 
-	return big.NewInt(0).Div(sum, big.NewInt(count))
+	return big.NewInt(0).Div(sum, big.NewInt(count+1))
```

Comment:
why is this `count+1`?

Reply:
This was initially added. Now removed! Thanks for noticing.